core/commands/config.go

```diff
@@ -140,11 +148,29 @@ Set the value of the 'Datastore.Path' key:
 	Type: ConfigField{},
 }
 
+// Returns bool to indicate if tested key is in the blocked scope.
+// (scope includes parent, direct, and child match)
+func inBlockedScope(testKey string, blockedScope string) bool {
```

Comment:
Why can't this function be implemented simply as:

return strings.HasPrefix(testKey, blockedScope)

Reply:
That returns false for testKey equal `Pinning`. Details in https://github.com/ipfs/go-ipfs/pull/7661#discussion_r537693938

Reply:
This implementation, as is, hides all of Pinning (there is nothing in addition to RemotePinning in there). Doing this can be accomplished with the one-liner above. So why use the much more complex version of inBlockedScope?

Reply:
I am assuming we would be fixing this right away in the MFS pinning PR.

Reply:
You are right, blocking entire `Pinning` for now would be enough.
I think this complex version aimed to work differently, but then we descoped MFS and I've found it can't be reused for `Identity.PrivKey` and we ended up with some scar tissue.

Feel free to remove/rewrite in MFS PR.